File: web/src/lib/mockData.ts
import type { BiomarkerWithSeries, DietEvent, FoodEntry } from "./api";

/** Representative mock data matching the real XLSX structure.
 *  Dates: 2023-09-15, 2024-05-31, 2025-03-13, 2026-05-22 (chronological).
 *  TODO: replace with live getBiomarkerResults() when auth is wired (Sprint 1 auth).
 */
export const MOCK_RESULTS: BiomarkerWithSeries[] = [
  // ── Lipids ──────────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-hdl",
      name_no: "P-HDL-kolesterol (High-Density Lipoprotein Cholesterol)",
      name_en: "HDL Cholesterol",
      unit: "mmol/L",
      ref_range_raw: "0,9 - 2,0",
      ref_low: 0.9,
      ref_high: 2.0,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 1.1, in_range: true },
      { tested_at: "2024-05-31", value: 1.5, in_range: true },
      { tested_at: "2025-03-13", value: 1.3, in_range: true },
      { tested_at: "2026-05-22", value: 1.3, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-ldl",
      name_no: "P-LDL-kolesterol (Low-Density Lipoprotein Cholesterol)",
      name_en: "LDL Cholesterol",
      unit: "mmol/L",
      ref_range_raw: "1,4 - 4,7",
      ref_low: 1.4,
      ref_high: 4.7,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 4.3, in_range: true },
      { tested_at: "2024-05-31", value: 3.5, in_range: true },
      { tested_at: "2025-03-13", value: 3.6, in_range: true },
      { tested_at: "2026-05-22", value: 4.1, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-chol",
      name_no: "P-Kolesterol (Cholesterol)",
      name_en: "Total Cholesterol",
      unit: "mmol/L",
      ref_range_raw: "3,3 - 6,9",
      ref_low: 3.3,
      ref_high: 6.9,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 5.5, in_range: true },
      { tested_at: "2024-05-31", value: 4.8, in_range: true },
      { tested_at: "2025-03-13", value: 4.8, in_range: true },
      { tested_at: "2026-05-22", value: 5.8, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-nonhdl",
      name_no: "P-non-HDL-kolesterol",
      name_en: "non-HDL Cholesterol",
      unit: "mmol/L",
      ref_range_raw: "1,5 - 4,8",
      ref_low: 1.5,
      ref_high: 4.8,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 4.4, in_range: true },
      { tested_at: "2024-05-31", value: 3.3, in_range: true },
      { tested_at: "2025-03-13", value: 3.5, in_range: true },
      { tested_at: "2026-05-22", value: 4.5, in_range: true },
    ],
  },

  // ── CBC ──────────────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-hgb",
      name_no: "B-Hemoglobin",
      name_en: "Hemoglobin",
      unit: "g/dL",
      ref_range_raw: "13,4 - 17,0",
      ref_low: 13.4,
      ref_high: 17.0,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2025-03-13", value: 15.4, in_range: true },
      { tested_at: "2026-05-22", value: 15.2, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-rbc",
      name_no: "B-Erytrocytter (Red Blood Cells)",
      name_en: "RBC",
      unit: "10¹²/L",
      ref_range_raw: "4,5 - 5,8",
      ref_low: 4.5,
      ref_high: 5.8,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2025-03-13", value: 5.4, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-wbc",
      name_no: "B-Leukocytter (White Blood Cells)",
      name_en: "WBC",
      unit: "10⁹/L",
      ref_range_raw: "3,9 - 9,8",
      ref_low: 3.9,
      ref_high: 9.8,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2025-03-13", value: 4.9, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-hct",
      name_no: "B-Hematokritt (Hematocrit)",
      name_en: "Hematocrit",
      unit: "%",
      ref_range_raw: "39 - 51",
      ref_low: 39,
      ref_high: 51,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2025-03-13", value: 45, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-mch",
      name_no: "Ery-MCH (Mean Corpuscular Hemoglobin)",
      name_en: "MCH",
      unit: "pg",
      ref_range_raw: "28 - 33",
      ref_low: 28,
      ref_high: 33,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2025-03-13", value: 29, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-mchc",
      name_no: "Ery-MCHC (Mean Corpuscular Hemoglobin Concentration)",
      name_en: "MCHC",
      unit: "g/dL",
      ref_range_raw: "32,8 - 36,6",
      ref_low: 32.8,
      ref_high: 36.6,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2025-03-13", value: 34.4, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-mcv",
      name_no: "Ery-MCV (Mean Corpuscular Volume)",
      name_en: "MCV",
      unit: "fL",
      ref_range_raw: "83 - 97",
      ref_low: 83,
      ref_high: 97,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2025-03-13", value: 83, in_range: true }],
  },

  // ── Metabolic ────────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-hba1c",
      name_no: "B-HbA1c (Glycated Hemoglobin)",
      name_en: "HbA1c",
      unit: "mmol/mol",
      ref_range_raw: "27 - 42",
      ref_low: 27,
      ref_high: 42,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2025-03-13", value: 34, in_range: true }],
  },

  // ── Thyroid ──────────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-tsh",
      name_no: "P-TSH (Thyroid Stimulating Hormone)",
      name_en: "TSH",
      unit: "mIU/L",
      ref_range_raw: "0,35 - 3,6",
      ref_low: 0.35,
      ref_high: 3.6,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 0.62, in_range: true },
      { tested_at: "2024-05-31", value: 0.46, in_range: true },
      { tested_at: "2025-03-13", value: 0.37, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-t4",
      name_no: "P-T4, fritt",
      name_en: "Free T4",
      unit: "pmol/L",
      ref_range_raw: "9,5 - 16,0",
      ref_low: 9.5,
      ref_high: 16.0,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 13.8, in_range: true },
      { tested_at: "2024-05-31", value: 12.4, in_range: true },
      { tested_at: "2025-03-13", value: 11.9, in_range: true },
    ],
  },

  // ── Renal ────────────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-creat",
      name_no: "P-Kreatinin (Creatinine)",
      name_en: "Creatinine",
      unit: "µmol/L",
      ref_range_raw: "60 - 105",
      ref_low: 60,
      ref_high: 105,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 68, in_range: true },
      { tested_at: "2024-05-31", value: 70, in_range: true },
      { tested_at: "2025-03-13", value: 66, in_range: true },
      { tested_at: "2026-05-22", value: 61, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-gfr",
      name_no:
        "Nyre-estimert GFR (Estimated Glomerular Filtration Rate) (overfl. = 1,73m²)",
      name_en: "eGFR",
      unit: "mL/min/1.73m²",
      ref_range_raw: "83 - 136",
      ref_low: 83,
      ref_high: 136,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 116, in_range: true },
      { tested_at: "2024-05-31", value: 113, in_range: true },
      { tested_at: "2025-03-13", value: 116, in_range: true },
      { tested_at: "2026-05-22", value: 116, in_range: true },
    ],
  },

  // ── Liver ────────────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-alat",
      name_no: "P-ALAT (Alanine Aminotransferase)",
      name_en: "ALT",
      unit: "U/L",
      ref_range_raw: "10 - 70",
      ref_low: 10,
      ref_high: 70,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 27, in_range: true },
      { tested_at: "2024-05-31", value: 22, in_range: true },
      { tested_at: "2025-03-13", value: 25, in_range: true },
      { tested_at: "2026-05-22", value: 55, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-gt",
      name_no: "P-GT",
      name_en: "GGT",
      unit: "U/L",
      ref_range_raw: "10 - 80",
      ref_low: 10,
      ref_high: 80,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2023-09-15", value: 18, in_range: true }],
  },

  // ── Nutrients ────────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-ferr",
      name_no: "P-Ferritin",
      name_en: "Ferritin",
      unit: "µg/L",
      ref_range_raw: "30 - 470",
      ref_low: 30,
      ref_high: 470,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 262, in_range: true },
      { tested_at: "2024-05-31", value: 217, in_range: true },
      { tested_at: "2025-03-13", value: 295, in_range: true },
      { tested_at: "2026-05-22", value: 335, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-folat",
      name_no: "P-Folat (Folate)",
      name_en: "Folate",
      unit: "nmol/L",
      ref_range_raw: "6,3 - 34",
      ref_low: 6.3,
      ref_high: 34,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 25, in_range: true },
      { tested_at: "2024-05-31", value: 25, in_range: true },
      { tested_at: "2025-03-13", value: 19, in_range: true },
      { tested_at: "2026-05-22", value: 14.4, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-b12",
      name_no: "P-Vit B12, total (Kobalamin) (Vitamin B12)",
      name_en: "Vitamin B12",
      unit: "pmol/L",
      ref_range_raw: "167 - 695",
      ref_low: 167,
      ref_high: 695,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 424, in_range: true },
      { tested_at: "2024-05-31", value: 497, in_range: true },
      { tested_at: "2025-03-13", value: 502, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-b12-active",
      name_no: "P-Vit B12 aktivt",
      name_en: "Active B12",
      unit: "pmol/L",
      ref_range_raw: "32 - 207",
      ref_low: 32,
      ref_high: 207,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2024-05-31", value: 128, in_range: true },
      { tested_at: "2025-03-13", value: 128, in_range: true },
      { tested_at: "2026-05-22", value: 150, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-vitd",
      name_no: "P-Vitamin D (25-OH)",
      name_en: "Vitamin D",
      unit: "nmol/L",
      ref_range_raw: "50 - 150",
      ref_low: 50,
      ref_high: 150,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2023-09-15", value: 70, in_range: true },
      { tested_at: "2024-05-31", value: 57, in_range: true },
      { tested_at: "2025-03-13", value: 43, in_range: false },
    ],
  },
  {
    biomarker: {
      id: "mock-transferrin",
      name_no: "P-Transferrin",
      name_en: "Transferrin",
      unit: "g/L",
      ref_range_raw: "1,9 - 3,3",
      ref_low: 1.9,
      ref_high: 3.3,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2023-09-15", value: 2.3, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-iron",
      name_no: "P-Jern",
      name_en: "Iron",
      unit: "µmol/L",
      ref_range_raw: "6,0 - 28",
      ref_low: 6.0,
      ref_high: 28,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2023-09-15", value: 18, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-homocys",
      name_no: "P-Homocystein",
      name_en: "Homocysteine",
      unit: "µmol/L",
      ref_range_raw: "6,2 - 16,0",
      ref_low: 6.2,
      ref_high: 16.0,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2023-09-15", value: 7.5, in_range: true }],
  },
  {
    biomarker: {
      id: "mock-mma",
      name_no: "P-MMA (Metylmalonsyre)",
      name_en: "MMA",
      unit: "µmol/L",
      ref_range_raw: "0,08 - 0,24",
      ref_low: 0.08,
      ref_high: 0.24,
      ref_type: "bounded",
    },
    series: [{ tested_at: "2023-09-15", value: 0.1, in_range: true }],
  },

  // ── Electrolytes ─────────────────────────────────────────────────────────────
  {
    biomarker: {
      id: "mock-na",
      name_no: "S-Natrium (Sodium)",
      name_en: "Sodium",
      unit: "mmol/L",
      ref_range_raw: "137 - 145",
      ref_low: 137,
      ref_high: 145,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2024-05-31", value: 139, in_range: true },
      { tested_at: "2025-03-13", value: 139, in_range: true },
    ],
  },
  {
    biomarker: {
      id: "mock-k",
      name_no: "S-Kalium (Potassium)",
      name_en: "Potassium",
      unit: "mmol/L",
      ref_range_raw: "3,5 - 5,0",
      ref_low: 3.5,
      ref_high: 5.0,
      ref_type: "bounded",
    },
    series: [
      { tested_at: "2024-05-31", value: 4.4, in_range: true },
      { tested_at: "2025-03-13", value: 4.3, in_range: true },
    ],
  },
];

/** Sample diet annotations for the signed-out correlation-overlay demo. */
export const MOCK_DIET_EVENTS: DietEvent[] = [
  {
    id: "mock-evt-carnivore",
    label: "Started carnivore",
    kind: "diet",
    started_on: "2024-05-31",
    ended_on: null,
    note: "Switched to an all-meat elimination diet.",
  },
  {
    id: "mock-evt-fast",
    label: "5-day fast",
    kind: "fast",
    started_on: "2025-03-13",
    ended_on: "2026-05-22",
    note: "Sample extended-fasting block.",
  },
];

/** Sample food-diary entries for the signed-out demo. */
export const MOCK_FOOD_ENTRIES: FoodEntry[] = [
  {
    id: "mock-food-1",
    logged_on: new Date().toISOString().slice(0, 10),
    meal: "breakfast",
    food_name: "Eggs, whole, cooked",
    brand: null,
    barcode: null,
    quantity_g: 100,
    energy_kcal: 155,
    carbs_g: 1.1,
    protein_g: 13,
    fat_g: 11,
    note: null,
  },
  {
    id: "mock-food-2",
    logged_on: new Date().toISOString().slice(0, 10),
    meal: "dinner",
    food_name: "Ribeye steak",
    brand: null,
    barcode: null,
    quantity_g: 250,
    energy_kcal: 728,
    carbs_g: 0,
    protein_g: 60,
    fat_g: 55,
    note: null,
  },
];

/** Derive summary stats from mock data. */
export function getMockStats() {
  const allDates = new Set(
    MOCK_RESULTS.flatMap((r) => r.series.map((p) => p.tested_at))
  );
  const sortedDates = [...allDates].sort();
  return {
    biomarkerCount: MOCK_RESULTS.length,
    panelCount: allDates.size,
    lastTestedAt: sortedDates.at(-1) ?? null,
  };
}

File: ios/EmpiricalTracker/Mock/MockData.swift
import BodyMetrics
import Core
import Foundation

/// Representative demo data used when the app runs in mock/demo mode.
/// Mirrors the shapes returned by the live backend so the dashboard and detail
/// views render without a network connection. Values approximate a person
/// 6 months into a carnivore diet — matches the app's primary use-case.
public enum MockData {

    // MARK: - Date helpers

    private static func daysAgo(_ n: Int) -> Date {
        Calendar.current.date(byAdding: .day, value: -n, to: Date())!
    }

    // MARK: - Biomarker series

    public static let biomarkerSeries: [BiomarkerWithSeries] = [
        // Lipids
        .make(
            id: "kolesterol", nameNo: "Kolesterol", nameEn: "Total Cholesterol", unit: "mmol/L",
            refLow: 0, refHigh: 5.0, refType: .lt,
            points: [(180, 5.8), (120, 6.1), (60, 6.4), (14, 6.6), (0, 6.7)]
        ),
        .make(
            id: "ldl", nameNo: "LDL-kolesterol", nameEn: "LDL Cholesterol", unit: "mmol/L",
            refLow: 0, refHigh: 3.0, refType: .lt,
            points: [(180, 3.5), (120, 3.9), (60, 4.2), (14, 4.4), (0, 4.5)]
        ),
        .make(
            id: "hdl", nameNo: "HDL-kolesterol", nameEn: "HDL Cholesterol", unit: "mmol/L",
            refLow: 1.0, refHigh: nil, refType: .gt,
            points: [(180, 1.4), (120, 1.6), (60, 1.9), (14, 2.1), (0, 2.2)]
        ),
        .make(
            id: "tg", nameNo: "Triglyserider", nameEn: "Triglycerides", unit: "mmol/L",
            refLow: 0, refHigh: 1.7, refType: .lt,
            points: [(180, 1.2), (120, 0.9), (60, 0.7), (14, 0.65), (0, 0.6)]
        ),
        .make(
            // Atherogenic-particle count — the lean-mass hyper-responder marker.
            id: "apob", nameNo: "ApoB", nameEn: "Apolipoprotein B", unit: "g/L",
            refLow: 0, refHigh: 1.2, refType: .lt,
            points: [(180, 0.95), (120, 1.05), (60, 1.15), (14, 1.20), (0, 1.25)]
        ),
        .make(
            // Genetic risk modifier — measured once, largely stable.
            id: "lpa", nameNo: "Lp(a)", nameEn: "Lipoprotein (a)", unit: "nmol/L",
            refLow: 0, refHigh: 75, refType: .lt,
            points: [(180, 42), (0, 44)]
        ),

        // Glycaemic / metabolic
        .make(
            id: "hba1c", nameNo: "HbA1c", nameEn: "HbA1c", unit: "mmol/mol",
            refLow: 0, refHigh: 48, refType: .lt,
            points: [(180, 38), (90, 36), (30, 34), (0, 33)]
        ),
        .make(
            // Rises with the high purine load of an all-meat diet (gout / renal load).
            id: "urat", nameNo: "Urat", nameEn: "Uric acid", unit: "µmol/L",
            refLow: 200, refHigh: 420, refType: .bounded,
            points: [(180, 330), (90, 360), (30, 390), (0, 410)]
        ),

        // Liver
        .make(
            id: "alt", nameNo: "ALT", nameEn: "ALT", unit: "U/L",
            refLow: 0, refHigh: 45, refType: .lt,
            points: [(90, 28), (30, 25), (0, 23)]
        ),
        .make(
            id: "ggt", nameNo: "GGT", nameEn: "GGT", unit: "U/L",
            refLow: 0, refHigh: 45, refType: .lt,
            points: [(90, 22), (30, 19), (0, 18)]
        ),

        // Renal
        .make(
            id: "kreatinin", nameNo: "Kreatinin", nameEn: "Creatinine", unit: "µmol/L",
            refLow: 60, refHigh: 105, refType: .bounded,
            points: [(90, 88), (30, 92), (0, 95)]
        ),

        // Nutrients
        .make(
            id: "ferritin", nameNo: "Ferritin", nameEn: "Ferritin", unit: "µg/L",
            refLow: 30, refHigh: 400, refType: .bounded,
            points: [(120, 85), (60, 110), (14, 130), (0, 145)]
        ),
        .make(
            id: "b12", nameNo: "B12", nameEn: "Vitamin B12", unit: "pmol/L",
            refLow: 145, refHigh: 569, refType: .bounded,
            points: [(90, 320), (30, 380), (0, 410)]
        ),
        .make(
            id: "vitd", nameNo: "Vitamin D", nameEn: "Vitamin D (25-OH)", unit: "nmol/L",
            refLow: 50, refHigh: 125, refType: .bounded,
            points: [(180, 42), (90, 65), (30, 78), (0, 82)]
        ),

        // Electrolytes (incl. the refeeding-syndrome pair watched when fasting)
        .make(
            id: "magnesium", nameNo: "Magnesium", nameEn: "Magnesium", unit: "mmol/L",
            refLow: 0.71, refHigh: 0.94, refType: .bounded,
            points: [(90, 0.84), (30, 0.82), (0, 0.81)]
        ),
        .make(
            id: "fosfat", nameNo: "Fosfat", nameEn: "Phosphate", unit: "mmol/L",
            refLow: 0.85, refHigh: 1.65, refType: .bounded,
            points: [(90, 1.10), (30, 1.05), (0, 1.00)]
        ),
    ]

    // MARK: - Diet events

    public static let dietEvents: [DietEvent] = [
        DietEvent(
            id: "evt-1",
            label: "Started carnivore diet",
            kind: .diet,
            startedOn: daysAgo(180),
            endedOn: nil,
            note: "Beef, eggs, organ meat only. Zero plants."
        ),
        DietEvent(
            id: "evt-2",
            label: "72-hour fast",
            kind: .fast,
            startedOn: daysAgo(120),
            endedOn: daysAgo(117),
            note: "Water + salt only."
        ),
        DietEvent(
            id: "evt-3",
            label: "Added vitamin D supplement",
            kind: .supplement,
            startedOn: daysAgo(90),
            endedOn: nil,
            note: "4000 IU daily."
        ),
    ]

    // MARK: - Food diary (Sprint 6)

    /// A representative carnivore-diet day: whole foods from Matvaretabellen / USDA plus
    /// one branded OFF product. Demonstrates source badges and sodium / saturated-fat totals.
    public static let foodEntries: [FoodEntry] = [
        FoodEntry(
            id: "fe-1", loggedOn: .now, meal: .breakfast,
            foodName: "Egg, whole, raw", quantityG: 150,
            energyKcal: 215, carbsG: 1.7, proteinG: 18.9, fatG: 15.0,
            saturatedFatG: 4.6, sodiumMg: 213, source: .usda
        ),
        FoodEntry(
            id: "fe-2", loggedOn: .now, meal: .breakfast,
            foodName: "Smør", brand: nil, quantityG: 20,
            energyKcal: 149, carbsG: 0.1, proteinG: 0.2, fatG: 16.4,
            saturatedFatG: 10.4, sodiumMg: 2, source: .mvt
        ),
        FoodEntry(
            id: "fe-3", loggedOn: .now, meal: .dinner,
            foodName: "Storfe, ribeye, rå", quantityG: 300,
            energyKcal: 873, carbsG: 0, proteinG: 62.4, fatG: 69.0,
            saturatedFatG: 28.5, sodiumMg: 156, source: .mvt
        ),
        FoodEntry(
            id: "fe-4", loggedOn: .now, meal: .snack,
            foodName: "Jarlsberg", brand: "TINE", barcode: "7038010009457", quantityG: 40,
            energyKcal: 137, carbsG: 0, proteinG: 10.8, fatG: 10.6,
            saturatedFatG: 6.9, sodiumMg: 244, source: .off
        ),
    ]

    // MARK: - Meal plans (Sprint 7)

    public static let mealPlans: [MealPlan] = [
        MealPlan(id: "mp-1", name: "Carnivore week", description: "Beef, eggs, butter — zero plants."),
        MealPlan(id: "mp-2", name: "Low-carb reset", description: "Under 30 g carbs/day."),
    ]

    /// A couple of planned meals across the current week, demonstrating per-day energy
    /// totals, the done toggle, plan grouping, and a free-text (no-macro) entry.
    public static let plannedMeals: [PlannedMeal] = [
        PlannedMeal(
            id: "pm-1", scheduledOn: Date(), meal: .breakfast,
            foodName: "Egg, whole, raw", quantityG: 150,
            energyKcal: 215, carbsG: 1.7, proteinG: 18.9, fatG: 15.0,
            done: false, planId: "mp-1"
        ),
        PlannedMeal(
            id: "pm-2", scheduledOn: Date(), meal: .dinner,
            foodName: "Storfe, ribeye, rå", quantityG: 300,
            energyKcal: 873, carbsG: 0, proteinG: 62.4, fatG: 69.0,
            done: false, planId: "mp-1"
        ),
        PlannedMeal(
            id: "pm-3", scheduledOn: Calendar.current.date(byAdding: .day, value: 1, to: Date())!,
            meal: .lunch, foodName: "Ribeye + eggs", note: "Free-text — no macros published.",
            done: false, planId: "mp-1"
        ),
    ]

    // MARK: - Body metrics (Sprint 8)

    /// A carnivore-diet weight-loss arc with periodic waist and blood-pressure readings,
    /// demonstrating each metric's own series (charts skip the rows where it is null) and
    /// the diet-event correlation overlay sharing the same timeline.
    public static let bodyMetrics: [BodyMetric] = [
        BodyMetric(id: "bm-1", measuredOn: daysAgo(180), weightKg: 92.4, waistCm: 104,
                   systolic: 138, diastolic: 88, note: "Baseline before carnivore."),
        BodyMetric(id: "bm-2", measuredOn: daysAgo(150), weightKg: 89.1, waistCm: 101),
        BodyMetric(id: "bm-3", measuredOn: daysAgo(120), weightKg: 86.7, waistCm: 98,
                   systolic: 131, diastolic: 84),
        BodyMetric(id: "bm-4", measuredOn: daysAgo(90), weightKg: 84.2, waistCm: 95,
                   systolic: 126, diastolic: 81),
        BodyMetric(id: "bm-5", measuredOn: daysAgo(60), weightKg: 82.0, waistCm: 92),
        BodyMetric(id: "bm-6", measuredOn: daysAgo(30), weightKg: 80.3, waistCm: 90,
                   systolic: 120, diastolic: 78),
        BodyMetric(id: "bm-7", measuredOn: daysAgo(0), weightKg: 79.1, waistCm: 89,
                   systolic: 118, diastolic: 76, note: "Down 13 kg."),
    ]

    // MARK: - Activity metrics (HealthKit — steps, active energy, exercise)

    public static let activityMetrics: [ActivityMetric] = (0..<90).map { i in
        let d = daysAgo(89 - i)
        let baseSteps = 7_500 + Int.random(in: -2_000...3_000)
        let energy = Double(baseSteps) * 0.045 + Double.random(in: -50...80)
        let exercise = max(0, baseSteps / 500 - 5 + Int.random(in: -5...15))
        return ActivityMetric(
            id: "am-\(i)",
            measuredOn: d,
            steps: baseSteps,
            activeEnergyKcal: energy,
            exerciseMinutes: exercise
        )
    }
}

// MARK: - Builder

private extension BiomarkerWithSeries {
    static func make(
        id: String,
        nameNo: String,
        nameEn: String,
        unit: String,
        refLow: Double?,
        refHigh: Double?,
        refType: BiomarkerInfo.RefType,
        points: [(daysAgo: Int, value: Double)]
    ) -> BiomarkerWithSeries {
        let info = BiomarkerInfo(
            id: id,
            nameNo: nameNo,
            nameEn: nameEn,
            unit: unit,
            refRangeRaw: buildRefRaw(low: refLow, high: refHigh, type: refType),
            refLow: refLow,
            refHigh: refHigh,
            refType: refType
        )
        let cal = Calendar.current
        let series: [ResultPoint] = points.map { (days, value) in
            let date = cal.date(byAdding: .day, value: -days, to: Date())!
            let inRange: Bool? = switch refType {
            case .bounded: (refLow.map { value >= $0 } ?? true) && (refHigh.map { value <= $0 } ?? true)
            case .lt:  refHigh.map { value <= $0 }
            case .gt:  refLow.map  { value >= $0 }
            case .none: nil
            }
            return ResultPoint(testedAt: date, value: value, inRange: inRange)
        }
        return BiomarkerWithSeries(biomarker: info, series: series.sorted { $0.testedAt < $1.testedAt })
    }

    private static func buildRefRaw(low: Double?, high: Double?, type: BiomarkerInfo.RefType) -> String {
        switch type {
        case .lt:      return "< \(high ?? 0)"
        case .gt:      return "> \(low ?? 0)"
        case .bounded: return "\(low ?? 0) - \(high ?? 0)"
        case .none:    return "—"
        }
    }
}
